### ISSUE TITLE
MSYS2 Mingw support

### ISSUE THREAD

**Issue by pal1000** [pal1000/mesa-dist-win#17]
MSYS2 Mingw is slightly faster than MSVC as it benefits from SSSE3 optimization via -march=core2 compiler flag which lacks equivalent in MSVC build and even further optimizations can be applied by changing march value to newer CPU generation. Historically @alexpux didn't post regular updates to MSYS2 Mingw builds of Mesa so I feel I should tackle this as well. Reverse engineering his build doesn't appear to be very hard.

### Tasks list
- [x] Support locating and loading of MINGW64 and MINGW32 environments;
- [x] Research necessary packages and patches and implement their management, use pre-built LLVM (info available here : https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-mesa) (patch is all that's left);
- [x] Update Mesa3D build to to use toolchain=mingw option;
- [x] Update distribution creation to support linux .a libraries;
- [x] pass necessary compiler and linker flags.

Note: Deployment tools work unchanged.

### Optional
- [ ] Self made LLVM build;
- [ ] swr build (needs upstream fixes).

### Longer term
- [ ] Meson build support.

**Comment by pal1000** [comment]
I took another look another look at this and it's easiest to mix up my batch scripts with most of PKGBUILD that MSYS2 uses. This should get things done really quickly.

**Comment by pal1000** [comment]
I drafted most of this and all I have to do is integrate it now. Toolchain selection menu is the hardest and most time consuming to implement but it's worth as I'll finally be able to support professional and enterprise editions of visual studio . 

This build also historically relied on an [out of tree patch](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-mesa/PKGBUILD) that was too aggressive to be accepted upstream as it was breaking MSVC build.  I tweaked that patch to avoid that and sent it [upstream](https://gitlab.freedesktop.org/mesa/mesa/merge_requests/362).
